### PR TITLE
Support ephemeral-storage resource config

### LINKED ISSUE #3307: Ephemeral storage is missed in proxy annotations and defaults are ignored
https://linkerd.io/2/reference/proxy-configuration/

Related to https://github.com/linkerd/website/issues/422

The following annotations are supported:

```
config.linkerd.io/proxy-memory-limit
config.linkerd.io/proxy-memory-request
config.linkerd.io/proxy-cpu-limit
config.linkerd.io/proxy-cpu-request
```

If a namespace has a configured [LimitRange](https://kubernetes.io/docs/concepts/policy/limit-range/) with `min` or `max` setting for `ephemeral-storage`, pods can't be started.

When other containers get default values from `defaultRequest` and `default` (limit) configurations of a LimitRange, Linkerd sidecar doesn't respect them for some reason and the ReplicaSet fails with the event:

```
  Warning  FailedCreate  49s (x6 over 2m26s)  replicaset-controller  (combined from similar events): Error creating: pods "tum-tururum-pum-pum-659f5f99cd-xpk4x" is forbidden: [minimum ephemeral-storage usage per Pod is 10Mi.  No request is specified., maximum ephemeral-storage usage per Pod is 10Gi.  No limit is specified.]

  Warning  FailedCreate      12s                replicaset-controller  Error creating: pods "tum-tururum-pum-pum-65f5788c57-v9dsc" is forbidden: maximum ephemeral-storage usage per Pod is 10Gi.  No limit is specified.
```

```
➜ linkerd version
Client version: stable-2.5.0
Server version: stable-2.5.0

➜ kubectl version --short
Client Version: v1.15.3
Server Version: v1.12.10-eks-2e569f
```

I would suggest to support custom annotations because some other third-parties can be installed into a K8s cluster that can prevent pods from starting. Maybe it is already possible somewhere in Linkerd's ConfigMap?

### PR DESCRIPTION
**Pull request by michaellzc**
The resource configuration does not support `ephemeral-storage`.

The [partials.resources](main/charts/partials/templates/_resources.tpl) named template should be updated to support such configuration.

The change can be validated by running under `linkerd2/viz/charts/linkerd-viz` directory

```bash
helm template --set prometheus.resources.ephemeral-storage.limit=4Gi .
```

```bash
helm template --set prometheus.resources.ephemeral-storage.request=4Gi .
```

```bash
helm template \
  --set prometheus.resources.ephemeral-storage.limit=4Gi \
  --set prometheus.resources.ephemeral-storage.request=4Gi .
```

Make sure it doesn't affect existing resources configuration

```bash
helm template --set prometheus.resources.cpu.limit=4Gi .
```

Fixes #3307

Signed-off-by: Michael Lin <mlzc@hey.com>

----------------
The convention linkerd chart uses to render resource is not so intuitive unless you read the docs or the template

In plain manifest

```yml
resources:
  limits:
    cpu: 2000m
  requests:
    cpu: 50m
```

in linkerd chart, the `resources` value is expecting something like, then they get map to places like `spec.containers[].resources.limits.cpu`, `spec.containers[].resources.limits.memory`

```yml
resources:
  cpu:
    request: 50m
    limit: 2000m
```

I get the intention to make the values easier to read for a specific resource (i.e. `cpu`, `memory`, `ephemeral-storage`).

The downside of this approach is the chart has to expose every possible resource type, and it only supports `cpu` and `memory` now. Maybe we should get rid of the mapping and just render `resources` as it is, so developers can just add whatever resource type as they wish without making changes to the linkerd chart. However, this would introduce a breaking change. (just an idea, not saying we should do it)

### CODE REVIEW

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 1]
```diff
@@ -85,6 +102,23 @@ jaeger:
...
+      # -- Maximum amount of memory that jaeger container can use
+      limit:
+      # -- Amount of memory that the jaeger container requests
+      request:
+    ephemeral-storage:
+      # -- Maximum amount of ephemeral storage that the jaeger container can use
+      limit: ""
+      # -- Amount of ephemeral storage that the jaeger container requests
+      request: ""
+
```
Comment (michaellzc) [inline jaeger/charts/linkerd-jaeger/values.yaml:121]: https://github.com/linkerd/linkerd2/blob/main/jaeger/charts/linkerd-jaeger/templates/tracing.yaml#L201-L203

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 2]
```diff
@@ -31,6 +31,23 @@ collector:
...
+      # -- Maximum amount of memory that collector container can use
+      limit:
+      # -- Amount of memory that the collector container requests
+      request:
+    ephemeral-storage:
+      # -- Maximum amount of ephemeral storage that the collector container can use
+      limit: ""
+      # -- Amount of ephemeral storage that the collector container requests
+      request: ""
+
```
Comment (michaellzc) [inline jaeger/charts/linkerd-jaeger/values.yaml:50]: https://github.com/linkerd/linkerd2/blob/main/jaeger/charts/linkerd-jaeger/templates/tracing.yaml#L114-L116

Used here but not present in the `values.yaml`

File: viz/charts/linkerd-viz/values.yaml [hunk 3]
```diff
@@ -301,7 +321,7 @@ grafana:
   logLevel: ""
   # -- log format (plain, json) of the grafana instance
   # @default -- defaultLogFormat
-  logFormat: ""  
+  logFormat: ""
```
Comment (michaellzc) [inline viz/charts/linkerd-viz/values.yaml:324]: I guess the last person who touched this line doesn't have `editorconfig` installed

https://github.com/linkerd/linkerd2/blob/main/.editorconfig#L7

**Comment by michaellzc** [comment]
All suggestions should have been addressed. Please let me know if I miss anything.

File: jaeger/charts/linkerd-jaeger/values.yaml (hunk 2 above)
Comment (alpeb) [reply jaeger/charts/linkerd-jaeger/values.yaml:50]: :+1:

File: jaeger/charts/linkerd-jaeger/values.yaml (hunk 1 above)
Comment (alpeb) [reply jaeger/charts/linkerd-jaeger/values.yaml:121]: :+1:

**Comment by alpeb** [comment]
Thanks @michaellzc , this is looking good :+1: 
Unit tests still need to be updated. You can do that with `go test ./... -update` and then commit the generated changes.

**Review by alpeb — APPROVED** [review]
Thanks @michaellzc :+1:

**Review by kleimkuhler — APPROVED** [review]
Looks great thanks @michaellzc!

**Comment by michaellzc** [comment]
Thanks.

Expecting another PR in the upcoming days for adding the annotations support as described here

https://github.com/linkerd/linkerd2/pull/7159#issuecomment-954403112

**Pull request merged**